core/src/test/java/io/grpc/internal/KeepAliveManagerTest.java

```diff
@@ -142,6 +141,8 @@ public void keepAlivePingDelayedByIncomingData() {
 
   @Test
   public void keepAlivePingFails() {
+    keepAliveManager = new KeepAliveManager(
+        new ClientKeepAlivePinger(transport), scheduler, ticker, 1000, 2000, false);
```

Comment:
Continue using pinger in this and similar tests. Only use `ClientKeepAlivePinger` in tests testing `ClientKeepAlivePinger`.

Basically you'd have about two new tests to test `ClientKeepAlivePinger` and nothing else would use it.